Products/CMFPlone/resources/browser/resource.py

```diff
@@ -31,39 +27,162 @@ class ResourceBase:
+                )
+
+        # register bundles from theme
+        theme_group_js = ResourceGroup(name="theme_js", group=root_group_js)
+        theme_group_css = ResourceGroup(name="theme_css", group=root_group_css)
+
+        # Collect theme data
+        themedata = {}
+        themedata["production_css"] = getattr(theme, "production_css", None)
+        themedata["development_css"] = getattr(theme, "development_css", None)
+        themedata["production_js"] = getattr(theme, "production_js", None)
+        themedata["development_js"] = getattr(theme, "development_js", None)
+
+        # add Theme JS
+        if themedata["production_js"]:
+            # we ignore development_js for external detection
+            external = themedata["production_js"].startswith("http")
+            resource = PloneScriptResource(
+                context=self.context,
+                name="theme",
+                depends="",
+                resource=(
+                    themedata["development_js"] or themedata["production_js"]
+                    if not external
+                    else None
+                ),
+                compressed=themedata["production_js"] if not external else None,
+                include=True,
+                unique=append_hash,
+                group=theme_group_js,
+                url=themedata["production_js"] if external else None,
+                crossorigin=None,
+                referrerpolicy=None,
+                type_=None,
+                integrity=not external,
+                nomodule=None,
+            )
+
+        # add Theme CSS
+        if themedata["production_css"]:
+            # we ignore development_css for external detection
+            external = themedata["production_css"].startswith("http")
+            theme_resource_css = PloneStyleResource(
+                context=self.context,
+                name="theme",
+                depends="",
+                resource=(
+                    themedata["development_css"] or themedata["production_css"]
+                    if not external
+                    else None
+                ),
+                compressed=themedata["production_css"] if not external else None,
+                include=True,
+                unique=append_hash,
+                group=theme_group_css,
+                url=themedata["production_css"] if external else None,
+                crossorigin=None,
+                referrerpolicy=None,
+                hreflang=None,
+                media="all",
+                rel="stylesheet",
+                sizes=None,
+                title=None,
+            )
+
+        resolver_js = ResourceResolver(root_group_js)
+        self.renderer_js = ResourceRenderer(
+            resolver_js, base_url=self.portal_state.portal_url()
+        )
+        resolver_css = ResourceResolver(root_group_css)
+        self.renderer_css = ResourceRenderer(
+            resolver_css, base_url=self.portal_state.portal_url()
+        )
```

Comment:
This function always resolves js and css, although the deriving viewlets use either js or css resources. Maybe this can be made more explicit by implemeting the plone resource iteration here and let a concrete resource factory implementation (e.g. ``create_resource_for`` or similar) on the subclass create the resource instances?

Reply:
Well, building up two trees with resources works fine here. What's the advantage of your approach?

Reply:
just nitpicking. it simply feels odd to build js resources if only css resources are needed and vice versa

Reply:
ah, ok, yes sure. if needed a future optimization can be done here.